kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/servlet/FileServlet.java

```diff
@@ -200,8 +200,11 @@ private void doGetIcon(HttpServletRequest req, HttpServletResponse resp) throws
                 }
                 MetaTypeInformation mti = mts.getMetaTypeInformation(b);
 
-                String[] pids = mti.getPids();
-                for (String p : pids) {
+                if (mti == null) {
```

Comment:
Considering that API for the method is not explicit on the case where bundle doesn't contain metatype info, this is probably a good check.